Add Zig design approach post (#5672)
sourcegraph/about: content/blogposts/2022/zig-programming-language-revisiting-design-approach.md

---
title: 'Revisiting the design approach to the Zig programming language'
authors:
  - name: Beyang Liu
    url: https://twitter.com/beyang
  - name: Justin Dorfman
    url: https://twitter.com/jdorfman
publishDate: 2022-08-16T10:00-07:00
description: 'Joe Chen is a Sourcegraph Software Engineer and the creator of Gogs, a painless self-hosted Git service. This is his story.”'
tags: [blog]
slug: 'zig-programming-language-revisiting-design-approach'
published: true
heroImage: https://storage.googleapis.com/sourcegraph-assets/blog/design-approach-zig-language-1200x675-2.jpg
socialImage: https://storage.googleapis.com/sourcegraph-assets/blog/design-approach-zig-language-1200x675-2.jpg
---

![](https://storage.googleapis.com/sourcegraph-assets/blog/design-approach-zig-language-1200x675-2.jpg)

C and C++ are everywhere, but the security issues and bugs caused by their lack of modern features are everywhere too (by some estimates, 80% of vulnerabilities are related to buffer overflows).

> Imagine could wave a magic wand over your C or C++ codebase and make it better? With Zig, you can.

Reviewer: ```suggestion
Imagine you could wave a magic wand over your C or C++ codebase and make it better? With Zig, you can.
```

> Imagine you could wave a magic wand over your C or C++ codebase and make it better? With Zig, you can.

Reviewer: ```suggestion
What if you could wave a magic wand over your C or C++ codebase and make it better? This is the promise of Zig.
```
